MyString.java

```diff
@@ -36,8 +41,22 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
-        return false;
+         boolean isSub = false;
+         if (str1.equals("")) {
+            return true;
+         }
+         for (int i = 0; i < str1.length(); i++) {
+            char c = str1.charAt(i);
+            int cTimes = countChar(str1, c);
+            if (countChar(str2, c) == cTimes) {
+                isSub = true;
+            } else if (countChar(str2, c) < cTimes) {
+                isSub = false;
+                break;
+            }

```

Comment:
missing case 
`countChar(str2, c) > cTimes`
could have written with == case and then switch to else in 53